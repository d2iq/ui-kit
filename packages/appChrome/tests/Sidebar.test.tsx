import React from "react";
import { shallow, mount } from "enzyme";
import { createSerializer } from "@emotion/jest";
import toJson from "enzyme-to-json";

import {
  Sidebar,
  SidebarItem,
  SidebarItemLabel,
  SidebarSection,
  SidebarSubMenuItem
} from "../";
import {
  SidebarSubMenuComponent,
  getSubItemList
} from "../components/SidebarSubMenu";
import { ProductIcons } from "../../icons/dist/product-icons-enum";

expect.addSnapshotSerializer(createSerializer());

describe("Sidebar", () => {
  it("renders", () => {
    const component = shallow(<Sidebar isOpen={true}>Sidebar content</Sidebar>);
    expect(toJson(component)).toMatchSnapshot();
  });
  it("calls onOpen callback", () => {
    const onOpenFn = jest.fn();
    const component = mount(
      <Sidebar isOpen={false} onOpen={onOpenFn}>
        Sidebar content
      </Sidebar>
    );
    expect(onOpenFn).not.toHaveBeenCalled();
    component.setProps({ isOpen: true });
    expect(onOpenFn).toHaveBeenCalled();
  });
  it("calls onClose callback", () => {
    const onCloseFn = jest.fn();
    const component = mount(
      <Sidebar isOpen={true} onClose={onCloseFn}>
        Sidebar content
      </Sidebar>
    );
    expect(onCloseFn).not.toHaveBeenCalled();
    component.setProps({ isOpen: false });
    expect(onCloseFn).toHaveBeenCalled();
  });

  describe("SidebarSection", () => {
    it("renders", () => {
      const component = shallow(
        <SidebarSection label="Label">
          <div>Sidebar section content</div>
        </SidebarSection>
      );
      expect(toJson(component)).toMatchSnapshot();
    });
  });

  describe("SidebarItem", () => {
    it("renders", () => {
      const onClickFn = jest.fn();
      const component = shallow(
        <Sidebar isOpen={true}>
          <SidebarItem
            icon={ProductIcons.ComponentsInverse}
            onClick={onClickFn}
            url="http://google.com"
            openInNewTab={true}
          >
            Item label
          </SidebarItem>
        </Sidebar>
      );
      expect(toJson(component)).toMatchSnapshot();
    });
  });

  describe("SidebarItemLabel", () => {
    it("renders", () => {
      const component = shallow(
        <SidebarItemLabel icon={ProductIcons.ComponentsInverse}>
          Item label
        </SidebarItemLabel>
      );
      expect(toJson(component)).toMatchSnapshot();
    });
  });

  describe("SidebarSubMenu", () => {
    const onClickFn = jest.fn();
    // tslint:disable:jsx-wrap-multiline
    const subMenuItems = [
      <SidebarSubMenuItem
        key={0}
        onClick={onClickFn}
        url="http://google.com"
        openInNewTab={true}
      >
        Item label
      </SidebarSubMenuItem>,
      <SidebarSubMenuItem
        key={1}
        onClick={onClickFn}
        url="http://google.com"
        openInNewTab={true}
      >
        Item label
      </SidebarSubMenuItem>
    ];
    // tslint:enable
    const component = shallow(
      <SidebarSubMenuComponent
        label={
          <SidebarItemLabel icon={ProductIcons.ComponentsInverse}>
            Item label
          </SidebarItemLabel>
        }
      >
        {subMenuItems}
      </SidebarSubMenuComponent>
    );
    it("renders", () => {
      expect(toJson(component)).toMatchSnapshot();
    });
    it("makes a list of submenu items", () => {
      const subItemResult = getSubItemList(subMenuItems);
      expect(subItemResult.props.children.length).toBe(2);
    });
  });

  describe("SidebarSubMenuItem", () => {
    it("renders", () => {
      const component = shallow(
        <SidebarItemLabel icon={ProductIcons.ComponentsInverse}>
          Item label
        </SidebarItemLabel>
      );
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
