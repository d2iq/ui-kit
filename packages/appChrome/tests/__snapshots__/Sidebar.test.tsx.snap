// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar SidebarItem renders 1`] = `
<ThemeProvider
  theme={[Function]}
>
  <StyledSidebar
    isOpen={true}
  >
    <SidebarItemComponent
      icon="product-components-inverse"
      onClick={[MockFunction]}
      openInNewTab={true}
      url="http://google.com"
    >
      Item label
    </SidebarItemComponent>
  </StyledSidebar>
</ThemeProvider>
`;

exports[`Sidebar SidebarItemLabel renders 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: auto;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
  padding-top: 12px;
  padding-bottom: 12px;
}

.emotion-0>div {
  width: auto;
}

.emotion-1 {
  box-sizing: border-box;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: initial;
  padding-right: 16px;
  line-height: 0;
}

.emotion-2 {
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <IconPropAdapter
      color="inherit"
      icon="product-components-inverse"
      size="s"
    />
  </div>
  <div
    className="emotion-2"
  >
    Item label
  </div>
</div>
`;

exports[`Sidebar SidebarSection renders 1`] = `
.emotion-0 {
  text-transform: capitalize;
  padding-left: 24px;
  padding-right: 24px;
  padding-bottom: 12px;
  padding-top: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: auto;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
  color: #828487;
  font-size: 14px;
  font-weight: 500;
  margin-bottom: 0;
  margin-top: 0;
}

.emotion-0>div {
  width: auto;
}

.emotion-1 {
  margin: 0;
  list-style: none;
  margin-left: 0;
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 0;
}

<div
  data-cy="sidebarSection"
>
  <h3
    className="emotion-0"
  >
    Label
  </h3>
  <ul
    className="emotion-1"
  >
    <div>
      Sidebar section content
    </div>
  </ul>
</div>
`;

exports[`Sidebar SidebarSubMenu renders 1`] = `
.emotion-0 {
  list-style: none;
  margin-left: 0;
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 0;
}

<ThemeProvider
  theme={[Function]}
>
  <li
    data-cy="sidebarSubMenu"
  >
    <Expandable
      indicatorPosition="right"
      label={
        <div>
          <SidebarItemLabel
            icon="product-components-inverse"
            iconWidth="s"
          >
            Item label
          </SidebarItemLabel>
        </div>
      }
      labelClassName="css-18glxmc"
    >
      <ul
        className="emotion-0"
      >
        <li
          key="0"
        >
          <SidebarSubMenuItem
            key=".$0"
            onClick={[MockFunction]}
            openInNewTab={true}
            url="http://google.com"
          >
            Item label
          </SidebarSubMenuItem>
        </li>
        <li
          key="1"
        >
          <SidebarSubMenuItem
            key=".$1"
            onClick={[MockFunction]}
            openInNewTab={true}
            url="http://google.com"
          >
            Item label
          </SidebarSubMenuItem>
        </li>
      </ul>
    </Expandable>
  </li>
</ThemeProvider>
`;

exports[`Sidebar SidebarSubMenuItem renders 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: auto;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
  padding-top: 12px;
  padding-bottom: 12px;
}

.emotion-0>div {
  width: auto;
}

.emotion-1 {
  box-sizing: border-box;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: initial;
  padding-right: 16px;
  line-height: 0;
}

.emotion-2 {
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <IconPropAdapter
      color="inherit"
      icon="product-components-inverse"
      size="s"
    />
  </div>
  <div
    className="emotion-2"
  >
    Item label
  </div>
</div>
`;

exports[`Sidebar renders 1`] = `
<ThemeProvider
  theme={[Function]}
>
  <StyledSidebar
    isOpen={true}
  >
    Sidebar content
  </StyledSidebar>
</ThemeProvider>
`;
